cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -161,7 +169,44 @@ func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperat
 	return string(apis.CStorPoolStatusInvalid), nil
 }
 
-// cStorPoolAddEvent does import of pool, and, if it fails, attemps to create pool based on CSP CR state
+// cStorPoolCreate does create of pool, and, if it fails, returns CreateFailed
+func (c *CStorPoolController) cStorPoolCreate(cStorPoolGot *apis.CStorPool) (string, error) {
+	if !IsCStorPoolCreateStatuses(cStorPoolGot) {
+		c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureCreate), string(common.MessageImproperPoolStatus))
+		return string(cStorPoolGot.Status.Phase), fmt.Errorf("invalid status %s for create pool %s", cStorPoolGot.Status.Phase, cStorPoolGot.Name)
+	}
+
+	devIDList, err := c.getDeviceIDs(cStorPoolGot)
+	if err != nil {
+		return string(apis.CStorPoolStatusCreateFailed), errors.Wrapf(err, "failed to get device id of disks for csp %s", cStorPoolGot.Name)
+	}
+
+	// ValidatePool is to check if pool attributes are correct.
+	err = pool.ValidatePool(cStorPoolGot, devIDList)
+	if err != nil {
+		c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureValidate), string(common.MessageResourceFailValidate))
+		return string(apis.CStorPoolStatusCreateFailed), err
+	}
+
+	// IsInitStatus is to check if initial status of cstorpool object is `init`.
```

Comment:
nit : comment does not look relevant.

Reply:
we will take in next one as travis takes more time

Reply:
I will do it in this PR itself